src/crimson/osd/pg_backend.cc

```diff
@@ -1015,12 +1018,22 @@ PGBackend::cmp_xattr_ierrorator::future<> PGBackend::cmp_xattr(
     if (result == 0) {
       logger().info("cmp_xattr: comparison returned false");
       osd_op.rval = -ECANCELED;
+      return crimson::ct_error::ecanceled::make();
+    } else if (result == -EINVAL) {
+      return crimson::ct_error::invarg::make();
```

Comment:
osd_op.rval = -EINVAL?

Reply:
Oh, still waiting on this one I think.  Otherwise LGTM

Reply:
remove osd_op.rval = -ECANCELED; it is not necessary here, error result will get from error_code,  osd_op_rval is used to save operation result.

Reply:
@athanatos any comments here?

Reply:
ok, sounds good